pkg/replicate/scheme.go

```diff
@@ -183,45 +168,20 @@ func newReplicationScheme(
 func (rs *replicationScheme) execute(ctx context.Context) error {
 	availableBlocks := []*metadata.Meta{}
 
-	level.Debug(rs.logger).Log("msg", "scanning blocks available blocks for replication")
-
-	if err := rs.fromBkt.Iter(ctx, "", func(name string) error {
-		rs.metrics.originIterations.Inc()
-
-		id, ok := thanosblock.IsBlockDir(name)
-		if !ok {
-			return nil
-		}
-
-		rs.metrics.originMetaLoads.Inc()
-
-		meta, metaNonExistentOrPartial, err := loadMeta(ctx, rs, id)
-		if metaNonExistentOrPartial {
-			// meta.json is the last file uploaded by a Thanos shipper,
-			// therefore a block may be partially present, but no meta.json
-			// file yet. If this is the case we skip that block for now.
-			rs.metrics.originPartialMeta.Inc()
-			level.Info(rs.logger).Log("msg", "block meta not uploaded yet. Skipping.", "block_uuid", id.String())
-			return nil
-		}
-		if err != nil {
-			return errors.Wrapf(err, "load meta for block %v from origin bucket", id.String())
-		}
+	metas, partials, err := rs.fetcher.Fetch(ctx)
+	if err != nil && metas == nil {
+		return err
```

Comment:
Should this error not be wrapped?

Reply:
Thanks, wrapping it seems better. I will update it later.

Reply:
Just checked it and seems it is not necessary to wrap it because the error is already wrapped in the baseFetcher. Other components (store, compact) don't wrap it as well.